### PR TITLE
Fixed a sporadic crash when stopping forever-monitor process.

### PR DESCRIPTION
**Pull request by jeremykross**
capture
capture1

@normanjoyner  - here are the two crashes I saw working on the environment variables today - I tend to think this isn't the fix for what you were seeing, but catching this error is safer in any case.

### CODE REVIEW

File: engines/docker.js [hunk 1]
```diff
@@ -271,6 +271,10 @@ module.exports = {
                                         });
                                     });
 
+                                    containers[container_id].on("error", function(error) {
+                                        core.loggers["containership.scheduler"].log("warn", `Caught an error managing the forever-monitor process for ${container_id}: ${error}`);
```
Comment (normanjoyner) [inline engines/docker.js]: @jeremykross while this should work given the node version we're using, can we just keep this consistent with the rest of scheduler and remove the template literals. we need to es6'ify the entire repository in the near future.

**Comment by normanjoyner** [comment]
@jeremykross looks good! can you sync with master?

**Comment by phildougherty** [comment]
👍

**Comment by jeremykross** [comment]
@normanjoyner I updated this code to set the container state in the error handler, but I think it's probably extraneous - if the follower doesn't crash when the error is thrown, the exit handler should run which sets the status to unloaded.  

Although, in my case, the error was relating to calling stop on an already stopped process, which means that the exit handler should already have run (which means the status should be unloaded).  Did you happen to grab a stack trace?

File: engines/docker.js [hunk 2]
```diff
@@ -5,6 +5,22 @@ var forever = require("forever-monitor");
...
+    commands.update_container(_.merge({
+        core: core
+        application_name: application_name,
+        container_id: container_id,
+        status: "unloaded",
+        host: null,
+        start_time: null,
+    }, should_respawn != undefined ? {respawn: should_respawn} : {}), function(err) {
+        if(err){
+            self.core.loggers["containership.scheduler"].log("warn", ["Failed to unload", application_name, "container:", container_id].join(" "));
```
Comment (nicktate) [inline engines/docker.js]: self.core does not exist, just use `core`

**Pull request merged**